certified-connectors/DocuSignDemo/script.csx

```diff
@@ -1785,6 +1785,39 @@ public class Script : ScriptBase
       acceptHeaderValue = "application/pdf";
     }
 
+    if ("GetDocumentsV2".Equals(this.Context.OperationId, StringComparison.OrdinalIgnoreCase))
+    {
+      var uriBuilder = new UriBuilder(this.Context.Request.RequestUri);
+      var newPath = uriBuilder.Path;
+      string[] documentDownloadOptions = { "Combined", "Archive", "Certificate", "Portfolio" };
+
+      acceptHeaderValue = "application/pdf";
+      string documentId = null;
+
+      foreach(var downloadOption in documentDownloadOptions)
+      {
+        if (newPath.Contains(downloadOption))
+        {
+          documentId = downloadOption;
+          break;
+        }
+      }
+
+      if (documentId == null)
+      {
+        uriBuilder.Path = newPath.Replace("/documentsDownload", "/");
+      }
+      else
+      {
+        uriBuilder.Path = newPath.Replace(newPath.Substring(
+          newPath.IndexOf(documentId),
+          newPath.IndexOf("/documentsDownload") + ("/documentsDownload".Length)- newPath.IndexOf(documentId)),
```

Comment:
Lets refactor this piece of code, because, using offsets can lead to future bugs.